rfc/0001-crashtracker-user-visable-log.md

```diff
@@ -0,0 +1,703 @@
+# RFC 0001: Crashtracker Structured Log Format (v1)
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in [IETF RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Summary
+This document describes version 1 of the crashinfo data format.
+
+## Motivation
+The `libdatadog` crashtracker detects program crashes.
+It automatically collects information relevant to the characterizing and debugging the crash, including stack-traces, the crash-type (e.g. SIGSIGV, SIGBUS, etc) crash, the library version, etc.
+This RFC establishes a standardized logging format for reporting this information.
+
+### Why structured json
+As a text-based format, json can be written to standard logging endpoints.
+It is (somewhat) human readable, so users can directly interpret the crash info off their log if necessary.
+As a structured format, it avoids the ambiguity of standard semi-structured stacktrace formats (as used by e.g. Java, .Net, etc).
+Due to the use of native extensions, it is possible for a single stack-trace to include frames from multiple languages (e.g. python may call C code, which calls Rust code, etc).
+Having a single structured format allows us to work across languages.
+
+## Proposed format
+A natural language description of the proposed json format is given here.
+An example is given in Appendix A, and the schema is given in Appendix B.
+
+### Required fields
+- `errortype`:
+    Currently, the only value is "crash", but this allows for extension to capture unhandled soft-errors, e.g. "panic", "uncaught exception", etc.
+- `incomplete`:
+    Boolean `false` if the crashreport is complete (i.e. contains all intended data), `true` if there is expected missing data.
+    This can happen becasue the crashtracker is architected to stream data to an out of process receiver, allowing a partial crash report to be emitted even in the case where the crashtracker itself crashed during stack trace collection.
+    This MUST be set to `true` if any required field is missing.
+- `metadata`:
```

Comment:
For consistency's sake, I propose matching the fields used here to align 1-to-1 with what we already use in instrumentation-telemetry, see [this](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/0cb13e7181acea60d4009d17c0a95fe6b6956e35/GeneratedDocumentation/ApiDocs/v2/development.md?plain=1#L89-L92) and [this](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/0cb13e7181acea60d4009d17c0a95fe6b6956e35/GeneratedDocumentation/ApiDocs/v2/development.md?plain=1#L105-L113). WDYT?

Reply:
Also, should these (i.e. `library_name`, `library_version`, `os_info` even be part of the structured json, given that they are already defined in the instrumentation-telemetry API which we are using to upload the crash reports? (see the `application` and `host` sections of the [required top level json keys](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/development.md#required-top-level-json-keys) in the instrumentation-telemetry api spec)